bson/doc.go

```diff
@@ -19,24 +20,96 @@
+// 		20. BSON DBPointer unmarshals to a primitive.DBPointer.
+// 		21. BSON symbol unmarshals to a primitive.Symbol.
 //
-// Example:
-// 		b, err := bson.Marshal(bson.D{{"foo", "bar"}})
-// 		if err != nil { return err }
-// 		var fooer struct {
-// 			Foo string
-// 		}
-// 		err = bson.Unmarshal(b, &fooer)
-// 		if err != nil { return err }
-// 		// do something with fooer...
+// The above mappings also apply when marshalling a D or M to BSON. Some other useful marshalling mappings are:
+//
+//       1. time.Time marshals to a BSON datetime.
+//       2. int8, int16, and int32 marshal to a BSON int32.
+//       3. int marshals to a BSON int32 if the value is between math.MinInt32 and math.MaxInt32, inclusive, and a BSON int64
+//       otherwise.
+//       4. int64 marshals to BSON int64.
+//       5. uint8 and uint16 marshal to a BSON int32.
+//       6. uint, uint32, and uint64 marshal to a BSON int32 if the value is between math.MinInt32 and math.MaxInt32,
+//       inclusive, and BSON int64 otherwise.
+//
+// Structs
+//
+// Structs can be marshalled/unmarshalled to/from BSON. When transforming structs to/from BSON, the following rules
+// apply:
+//
+//     1. Only exported fields in structs will be marshalled or unmarshalled.
+//
+//     2. When marshalling a struct, each field will be lowercased to generate the key for the corresponding BSON element.
+//     For example, a struct field named "Foo" will generate key "foo". This can be overriden via a struct tag (e.g.
+//     `bson:"fooField"` to generate key "fooField" instead).
+//
+//     3. An embedded struct field is marshalled as a subdocument. The key will be the lowercased name of the field's type.
+//
+//     4. A pointer field is marshalled as the underlying type if the pointer is non-nil. If the pointer is nil, it is
+//     marshalled as a BSON null value.
+//
+//     5. When unmarshalling, a field of type interface{} will follow the D/M type mappings listed above. BSON documents
+//     unmarshalled into an interface{} field will be unmarshalled as a D.
+//
+// The following struct tags can be used to configure behavior:
+//
+//     1. omitempty: If the omitempty struct tag is specified on a field, the field will not be marshalled if it is set to
+//     the zero value. By default, a struct field is only considered empty if the field's type implements the Zeroer
+//     interface and the IsZero method returns true. If the field's type does not implement Zeroer, the field will be
+//     marshalled as an embedded document.
+//
+//     2. minsize: If the minsize struct tag is specified on a field of type int64, uint, uint32, or uint64 and the value of
+//     the field can fit in a signed int32, the field will be serialized as a BSON int32 rather than a BSON int64. For other
+//     types, this tag is ignored.
+//
+//     3. truncate: If the truncate struct tag is specified on a field with a non-float64 numeric type, BSON doubles unmarshalled
```

Comment:
What about float32?

Reply:
This was supposed to be for float32 and float64, not just float64. Fixed to say "non-float numeric types"